### PR TITLE
When WebGL contexts are lost the extension are invalidated

### PR DESCRIPTION
**Pull request by Zekian**
When a [webgl context is lost extensions are invalidated](https://www.khronos.org/registry/webgl/specs/latest/1.0/#5.14.14), and extensions such as 'OES_vertex_array_object' can not be reinitialized via `emscripten_webgl_enable_extension` since variables like `GLctx['createVertexArray']` must be reintialized.

In the webgl callback handler, I have marked extensions as not initialized for the webglcontextlost event and  added a call to `GL.initExtension` for the webglcontextrestored event.

This way extensions are correctly initialized like in context creation.

### CODE REVIEW

File: src/library_html5.js [hunk 1]
```diff
@@ -993,6 +993,11 @@ var LibraryJSEvents = {
       var handlerFunc = function(event) {
         var e = event || window.event;
 
+        if(target.GLctxObject) {
```
Comment (kripken) [inline src/library_html5.js]: spaces after `if`

**Comment by kripken** [comment]
@juj, what do you think? You know this stuff the best.

**Comment by Zekian** [comment]
Would you like me to create a small example demonstrating the bug that this fixes?

I'll correct the style.

**Comment by juj** [comment]
Sorry for the delay here.

We definitely have a bug here that needs correcting. However let's see if we can do it without introducing a new state boolean? If you do have a test that we could add to the suite for context loss and restore, that would be very useful!

**Comment by juj** [comment]
Oh, actually, nm the above comment about the state boolean, re-reading I notice that `target.GLctxObject.initExtensionsDone` is a variable that already exists from before, so it's all good.

Though the location of where the hooking into the context loss and restore events occurs is misplaced, since it's hooking only if user registers to context loss and restore in the first place, and only if user is using the HTML5 API (user could be using something else, like the EGL API). It would be better to hook into context loss and restore after the try-catch here: https://github.com/kripken/emscripten/blob/incoming/src/library_gl.js#L437 , then the logic would be run independent of what GL helper API one is using.

**Pull request closed without merging**